### PR TITLE
Consolidate

### PR DESCRIPTION
**Pull request by akremstudy**
### Summary

Made quite a few edits:
- removed deprecated TellorX mock solidtity contracts and SampleXReporter
- Edited the SampleFlexReporter solidity contract to work and fixed test, this wasn't working before
- Added base_fee flag to allow user to input that in the cli if they wish (its another way to estimate the max fee)
- Added a request-withdraw and a stake command to be able to call the respective methods in the oracle contract separate from the reporter command.
- Separate common command options into two [See here](https://github.com/tellor-io/telliot-feeds/blob/e832beb7787d53716c4de5768447237306c5d77f/src/telliot_feeds/cli/utils.py#L294-L428)
- Decreased the max priority fee (aka tip) default range to 3 gwei since 80 is too high.
- Adjust Diva reporter to use GasFees object to generate gas fields when reporting and also consolidate the code to only adapt the method it needs instead of overwriting the report_once and report methods.
- Add stake command
- Same as diva with Flashbots reporter
- Remove redundant code in RNG reporter.
- Added GasFees class that would be inherited and used to generated gas params for the reporter.
- Add StakeInfo class to be used in the reporter to monitor when stake amount is changed or reporter is slashed currently the reporter will auto stake if any of these events happen.
- Broke up some of the code in the reporter for easier readability and usage
### Steps Taken to QA Changes

- Added new tests for the new classes plus another reporter test to check the changes
- Consolidated the previous reporter tests into one
- Fixed tests that were broken due to the changes

Example transactions using cli:
- [withdraw txn](https://sepolia.etherscan.io/tx/0xbe56923d96b794148ce5069f533e07041fa9b1172ce58b8aedacc051891a6998)
- [stake txn](https://sepolia.etherscan.io/tx/0x33665fb687fc172f5f440374dfcc9fecda9c746e6c4b147a96cd58f1fbdda337)
- [request withdraw](https://sepolia.etherscan.io/tx/0xb4af86a48cbee84c321078af188703c62995420905ac3449d2d699d3b17f0bd3)
- [report](https://sepolia.etherscan.io/tx/0x5790690624c81a349a1ae145bc5710cf4dbefeed6a3cac9e5f78814375906d9c)
### Checklist

This pull request is:

- [ ] A documentation error, docs update, or typographical error fix
	- No tests or issue needed
- [x] A code fix
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. Fixes without tests will not be accepted unless it's related to the documentation only.
    - Please make sure docs are updated if need be
- [x] A feature implementation
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure docs updates are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Comment by akremstudy** [comment]
@oraclown, Can I get a review on this pr specifically these two files [reporter](https://github.com/tellor-io/telliot-feeds/blob/e832beb7787d53716c4de5768447237306c5d77f/src/telliot_feeds/reporters/tellor_360.py) and [GasFees](https://github.com/tellor-io/telliot-feeds/blob/e832beb7787d53716c4de5768447237306c5d77f/src/telliot_feeds/reporters/gas.py).  Made a lot of changes but those are the main ones, if you can review them first then I can make changes before you spend time on the other files.

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks, @akremstudy ! Awesome refactoring of the reporter. Huge. I has some minor suggestions. Some tests needed for the new cli commands. I'd also encourage posting some example txs using the new cli commands, as well as just reporting, just as a sanity check. Great cli additions and updates to gas fees calculation code!

File: src/telliot_feeds/cli/commands/request_withdraw_stake.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    max_priority_fee_range: int,
+) -> None:
+    """Request withdraw of tokens from oracle"""
+    ctx.obj["ACCOUNT_NAME"] = account_str
+
+    accounts = get_accounts_from_name(account_str)
+    if not accounts:
+        return
+
+    ctx.obj["CHAIN_ID"] = accounts[0].chains[0]  # used in reporter_cli_core
```
[the remark this answers is not in the document]
Comment (akremstudy) [reply src/telliot_feeds/cli/commands/request_withdraw_stake.py:52]: if no accounts it exits

File: src/telliot_feeds/reporters/tellor_360.py [hunk 2]
```diff
@@ -1,63 +1,110 @@
...
+        self.chain_id = chain_id
+        self.acct_addr = to_checksum_address(self.account.address)
+        logger.info(f"Reporting with account: {self.acct_addr}")
+
+    async def get_stake_amount(self) -> Tuple[Optional[int], ResponseStatus]:
+        """Reads the current stake amount from the oracle contract
 
-        assert self.acct_addr == to_checksum_address(self.account.address)
+        Returns:
+        - (int, ResponseStatus) the current stake amount in TellorFlex
```
[the remark this answers is not in the document]
Comment (akremstudy) [reply src/telliot_feeds/reporters/tellor_360.py:77]: yeah

File: src/telliot_feeds/utils/stake_info.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    def is_in_dispute(self) -> bool:
+        """Check if staker has been disputed"""
+        if len(self.staker_balance_history) == self.max_data:
+            if self.staker_balance_history[-1] < self.staker_balance_history[-2]:
+                logger.warning("Your staked balance has decreased, account might be in dispute")
+                return True
+        return False
+
+    def stake_amount_change(self) -> bool:
```
[the remark this answers is not in the document]
Comment (akremstudy) [reply src/telliot_feeds/utils/stake_info.py:60]: same

File: src/telliot_feeds/utils/stake_info.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    @property
+    def last_report_time(self) -> int:
+        """Return report count"""
+        return self.last_report
+
+    def is_in_dispute(self) -> bool:
+        """Check if staker has been disputed"""
+        if len(self.staker_balance_history) == self.max_data:
+            if self.staker_balance_history[-1] < self.staker_balance_history[-2]:
```
[the remark this answers is not in the document]
Comment (akremstudy) [reply src/telliot_feeds/utils/stake_info.py:55]: it shouldn't since it checks if list length is 2 first.

**Comment by 0xSpuddy** [comment]
Testing by running a reporter on each testnet listening for tips / reporting at regular intervals. I haven't tested every combination of flags, but no errors so far.

**Review by oraclown — APPROVED** [review]
Thanks for the updates, @akremstudy !

**Pull request merged**